Source/WTF/wtf/win/RunLoopWin.cpp

```diff
@@ -170,7 +179,8 @@ void RunLoop::TimerBase::start(Seconds interval, bool repeat)
     m_isActive = true;
     m_interval = interval;
     m_nextFireDate = MonotonicTime::timePointFromNow(m_interval);
-    ::SetTimer(m_runLoop->m_runLoopMessageWindow, bitwise_cast<uintptr_t>(this), interval.millisecondsAs<UINT>(), nullptr);
+    m_runLoop->m_liveTimers.add(bitwise_cast<uintptr_t>(this));
```

Comment:
Is this accessing `m_liveTimers` from multiple threads without locking?

Reply:
`m_runLoop->m_loopLock` is held here and inside `RunLoop::TimerBase::stop` when we call `m_liveTimers.remove`.

The `m_runLoop->m_loopLock` isn't held when we do the `m_liveTimers.contains` check inside wndProc. I could take the `m_runLoop->m_loopLock` there as well - if it's live we take it after we call timerFired() anyway.